src/redux/books/books.js

```diff
@@ -0,0 +1,18 @@
+const ADD = 'ADD';
+const REMOVE = 'REMOVE';
```

Comment:
- [ ] Good job so far From but from [ducks pattern](https://github.com/erikras/ducks-modular-redux)...` const CREATE = 'my-app/widgets/CREATE';` so you should have something like this `const ADD_BOOK = bookstore/books/ADD_BOOK ` and  `const REMOVE_BOOK = bookstore/books/REMOVE_BOOK`. Please fix it to make your code works as expected :smile: 

#### Find an example below :arrow_down: 

```
const ADD_BOOK = bookstore/books/ADD_BOOK ;
const REM_BOOK = bookstore/books/REM_BOOK;

// Reducer Section
export default function reducer(state = [ ], action = {}) {
  switch (action.type) {
    case ADD_BOOK:
      return [
        ...state,
        action.payload,
      ];
    case REM_BOOK:
      return [
        state.filter((item) => item !== action.payload),
      ];
    default:
      return state;
  }
}

// Action Creators Section
export function addBook(obj) {
  return { type: ADD_BOOK, payload: obj };
}

export function removeBook(obj) {
  return { type: ADD_BOOK, payload: obj };
}

```

Reply:
ok i change the duck pattern that you told me to put it